tree.go

```diff
@@ -958,41 +948,77 @@ func (n *LeafNode) updateCn(index byte, value []byte, c *Point) {
 }
 
 func (n *LeafNode) updateLeaf(index byte, value []byte) {
-	c, oldc := n.getOldCn(index)
-
+	// Update the corresponding C1 or C2 commitment.
+	var c *Point
+	var oldC Point
+	if index < NodeWidth/2 {
+		c = n.c1
+		oldC = *n.c1
+	} else {
+		c = n.c2
+		oldC = *n.c2
+	}
 	n.updateCn(index, value, c)
 
-	n.updateC(index, c, oldc)
+	// Batch the Fr transformation of the new and old CX.
+	var frs [2]Fr
+	toFrMultiple([]*Fr{&frs[0], &frs[1]}, []*Point{c, &oldC})
+
+	// If index is in the first NodeWidth/2 elements, we need to update C1. Otherwise, C2.
+	cxIndex := 2 + int(index)/(NodeWidth/2) // [1, stem, -> C1, C2 <-]
+	n.updateC(cxIndex, frs[0], frs[1])
 
 	n.values[index] = value
 }
```

Comment:
Single leaf node updating new batching:
- First capture corresponding `c` and `oldC`.
- Batch Fr calc on both, since before we did two separate `toFr`.
- L968 and surrounding comments are paranoid explicit on "CX index" logic.